modules/javafx.web/src/main/java/com/sun/javafx/webkit/prism/WCFontImpl.java

```diff
@@ -126,6 +126,9 @@ private FontStrike getFontStrike()
     }
 
     @Override public int[] getGlyphCodes(char[] chars) {
+        // Calling TextLayout.getRuns will lead to loading correct fallback
+        // font from native font engine.
+        TextUtilities.createLayout(new String(chars), getPlatformFont()).getRuns();
```

Comment:
Will there be a performance impact (e.g., will this cause all text strings to be laid out twice?)

Reply:
There won't be any performance impact because the above method only be called by the WebCore::GlyphPage for a given character sequence

> Will there be a performance impact (e.g., will this cause all text strings to be laid out twice?)

There won't be any performance impact because the above method only be called by the WebCore::GlyphPage for a given character sequence. It is mainly used to check whether the font supports the given unicode char point, and based on that it will decide how to break the text. Also this method will be called only once during the entire life of WebKit for the char sequence + Font + size + weight combination.